doc/flame/collision_detection.md

```diff
@@ -120,6 +120,41 @@ other way around. If there are no intersections with the edges on a solid hitbox
 position is instead returned.
 
 
+### Collision order
+
+If a `PositionComponent` collides with more than one other object within a given time step, then
+the `onCollision` callbacks will be called in an essentially random order.  In some cases this can
+be a problem, such as in a bouncing ball game where the trajectory of the ball can differ depending
+on which other object was hit first.  To help resolve this the `collisionsCompletedNotifier`
+listener can be used - this triggers at the end of the collision detection process.
+
+An example of how this might be used is to add a local variable in your `PositionComponent` to save
+the other components with which this one is colliding:
+`List<PositionComponent> collisionComponents = [];`.  The `onCollision` callback is then used to
+save all the other `PositionComponent`s to this list:
+
+```dart
+@override
+void onCollision(Set<Vector2> intersectionPoints, PositionComponent other) {
+  collisionComponents.add(other);
+  super.onCollision(intersectionPoints, other);
+}
+
+```
+
+Finally, one adds a listener to the `onLoad` method of the `PositionComponent` to call a routine
+which will resolve how the collisions should be dealt with:
+
+```dart
+(game as HasCollisionDetection).collisionDetection.collisionsCompletedNotifier.
+                                addListener(() {
+  resolveCollisions();
+});
```

Comment:
Formatting

Reply:
I have to break the line (as it's too long otherwise).  I've now done it in a way that looks nice to me, happy to hear your thoughts.

Reply:
if you put it in an IDE and or https://dartpad.dev it will correctly format it for you, since it follows 80 width by default

Reply:
Couldn't get VScode, dartpad or tutorialspoint to do it automatically for me. :-(